src/Security/Authentication/OAuth/src/OAuthHandler.cs

```diff
@@ -91,7 +91,12 @@ protected override async Task<HandleRequestResult> HandleRemoteAuthenticateAsync
                     failureMessage.Append(";Uri=").Append(errorUri);
                 }
 
-                return HandleRequestResult.Fail(failureMessage.ToString(), properties);
+                var ex = new Exception(failureMessage.ToString());
+                ex.Data["error"] = error.ToString();
+                ex.Data["error_description"] = errorDescription.ToString();
```

Comment:
IIRC, `StringValues.ToString()` returns an empty string, so you could potentially end up with an empty entry in the data dictionary. Would that be worth refactoring to only add `error_description`/`error_uri` when they are present in the query string?

Reply:
As part of the automatic `access_denied` error handling, we also call `HandleRequestResult.Fail` at line 78. Would it make sense to do a similar thing to preserve the error description/uri?

Reply:
I'm not too worried about the empty strings, they're harmless and omitting them would complicate the code.